src/app/content/components/TableOfContents/index.tsx

```diff
@@ -23,28 +24,171 @@ interface SidebarProps {
   page?: Page;
 }
 
+function TabTrapper({
+  mRef,
+  isTocOpen,
+}: {
+  mRef: MutableRefObject<HTMLElement>;
+  isTocOpen: boolean;
+}) {
+  const isPhone = useMatchMobileMediumQuery();
+  const isMobile = useMatchMobileQuery();
+
+  React.useEffect(() => {
+    const refIsSensible = mRef && 'current' in mRef && mRef.current;
+
+    if (!refIsSensible) {
+      return;
+    }
+    const otherRegions =
+      document?.querySelectorAll(
+        '[data-testid="navbar"],[data-testid="bookbanner"]'
```

Comment:
I'm not sure we want to use testids for functionality, I usually think of these as being free to change for testing purposes.